src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnectionFactory.cs

```diff
@@ -39,7 +39,7 @@ override protected DbConnectionInternal CreateConnection(DbConnectionOptions opt
             SqlInternalConnection result = null;
             SessionData recoverySessionData = null;
 
-            SqlConnection sqlOwningConnection = owningConnection as SqlConnection;
+            SqlConnection sqlOwningConnection = (SqlConnection)owningConnection;
```

Comment:
@saurabh500 If this line is kept as is, then code behavior changes. The changes I suggest are readability changes, no behavior changes. Is such behavior change needed? Won't it facilitate error hiding?